x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/latency_alerts_history_chart.tsx

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import moment from 'moment';
+import { EuiPanel, EuiSpacer } from '@elastic/eui';
+import { EuiFlexGroup } from '@elastic/eui';
+import { EuiFlexItem } from '@elastic/eui';
+import { EuiTitle } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { EuiText } from '@elastic/eui';
+import { useFetchTriggeredAlertsHistory } from '@kbn/observability-plugin/public';
+import {
+  AnnotationDomainType,
+  LineAnnotation,
+  Position,
+} from '@elastic/charts';
+import { EuiIcon } from '@elastic/eui';
+import { EuiBadge } from '@elastic/eui';
+import { getDurationFormatter } from '../../../../../common/utils/formatters';
+import { getLatencyChartSelector } from '../../../../selectors/latency_chart_selectors';
+import { LatencyAggregationType } from '../../../../../common/latency_aggregation_types';
+import { useFetcher } from '../../../../hooks/use_fetcher';
+import { TimeseriesChart } from '../../../shared/charts/timeseries_chart';
+import { filterNil } from '../../../shared/charts/latency_chart';
+import {
+  getMaxY,
+  getResponseTimeTickFormatter,
+} from '../../../shared/charts/transaction_charts/helper';
+
+interface LatencyAlertsHistoryChartProps {
+  serviceName: string;
+  start: string;
+  end: string;
+  transactionType?: string;
+  latencyAggregationType: LatencyAggregationType;
+  environment: string;
+  timeZone: string;
+  ruleId: string;
+}
+export function LatencyAlertsHistoryChart({
+  serviceName,
+  start,
+  end,
+  transactionType,
+  latencyAggregationType,
+  environment,
+  timeZone,
+  ruleId,
+}: LatencyAlertsHistoryChartProps) {
+  const { data, status } = useFetcher(
+    (callApmApi) => {
+      if (
+        serviceName &&
+        start &&
+        end &&
+        transactionType &&
+        latencyAggregationType
+      ) {
+        return callApmApi(
+          `GET /internal/apm/services/{serviceName}/transactions/charts/latency`,
+          {
+            params: {
+              path: { serviceName },
+              query: {
+                environment,
+                kuery: '',
+                start: moment().subtract(30, 'days').toISOString(),
+                end: moment().toISOString(),
+                transactionType,
+                transactionName: undefined,
+                latencyAggregationType,
+              },
+            },
+          }
+        );
+      }
+    },
+    [
+      end,
+      environment,
+      latencyAggregationType,
+      serviceName,
+      start,
+      transactionType,
+    ]
+  );
+  const memoizedData = useMemo(
+    () =>
+      getLatencyChartSelector({
+        latencyChart: data,
+        latencyAggregationType,
```

Comment:
Why wouldn't you want to rerun the function when `latencyAggregationType` changes?

Reply:
@CoenWarmer For this part, I'm just following how APM has implemented it.

Reply:
Ok, maybe @sqren knows?

Reply:
I don't recall. 

@cauemarcondes Afaict you implemented this initially in https://github.com/elastic/kibana/pull/84634/files#diff-f615240301482e0d4e5d73f0ccd406fea462cd7493e857c0f5225e3b61dcd1c0R70.